cli/src/pcluster/config/cluster_config.py

```diff
@@ -1755,17 +1865,37 @@ def queue_ami(self):
 
 
 class SlurmQueue(_CommonQueue):
-    """Represent the Slurm Queue resource."""
+    """Represents a Slurm Queue that has Compute Resources with both Single and Multiple Instance Types."""
 
     def __init__(
         self,
-        compute_resources: List[SlurmComputeResource],
+        compute_resources: List[_BaseSlurmComputeResource],
         networking: SlurmQueueNetworking,
+        allocation_strategy: str = "lowest-price",
         **kwargs,
     ):
         super().__init__(**kwargs)
         self.compute_resources = compute_resources
         self.networking = networking
+        if any(isinstance(compute_resource, SlurmFlexibleComputeResource) for compute_resource in compute_resources):
+            self.allocation_strategy = allocation_strategy
+
+    @property
+    def instance_type_list(self):
+        """Return the list of instance types associated to the Queue."""
+        instance_types = set()
+        for compute_resource in self.compute_resources:
+            instance_types.update(compute_resource.instance_types)
+        return list(instance_types)
+
+    @property
+    def instance_types_with_instance_storage(self):
+        """Return a set of instance types in the queue that have instance store."""
+        result = set()
+        for instance_type in self.instance_type_list:
+            if SlurmFlexibleComputeResource.get_instance_type_info(instance_type).instance_storage_supported():
```

Comment:
I think we shouldn't use the SlurmFlexibleComputeResource here.
What about having a method to get the list of instance type info, something like instance_type_info_list?
This method will be exposed by the compute_resource, by both the SlurmFlexibleComputeResource and SlurmComputeResource, but in this way, the caller (instance_types_with_instance_storage) doesn't have to know about them.

Reply:
I agree, the `Queue` Class shouldn't rely on the `get_instance_type_info` method from `SlurmFlexibleComputeResource`.
I'll push the responsibility to obtain the list of instance_types that have instance store down to the compute resources as suggested.